src/Microsoft.AspNetCore.Mvc.Core/Internal/PhysicalFileResultExecutor.cs

```diff
@@ -44,15 +63,32 @@ public Task ExecuteAsync(ActionContext context, PhysicalFileResult result)
                     count: null,
                     cancellation: default(CancellationToken));
             }
-            else
+            else if (range == null || !fileStream.CanSeek)
             {
-                var fileStream = GetFileStream(result.FileName);
-
                 using (fileStream)
```

Comment:
GetFileStream and using (fileStream) need to stay together. Otherwise you miss closing the stream in a few places. You moved GetFileStream to avoid duplication for the last else block, but you end up opening the stream when you didn't need to for the (rangeLength == 0) scenario and forgetting to close it. Copy the GetFileStream call to your last else block and add a using statement there too. Even better: `using (var fileStream = GetFileStream(result.FileName))`